x-pack/plugins/entity_manager/server/lib/v2/definitions/setup_entity_definitions_index.ts

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { IClusterClient } from '@kbn/core/server';
+import { DEFINITIONS_ALIAS, TEMPLATE_VERSION } from '../constants';
+
+const definitionsIndexTemplate = {
+  name: `${DEFINITIONS_ALIAS}-template`,
+  index_patterns: [`${DEFINITIONS_ALIAS}-*`],
+  _meta: {
+    description: "Index template for the Elastic Entity Model's entity definitions index.",
+    managed: true,
+    managed_by: 'elastic_entity_model',
+  },
+  version: TEMPLATE_VERSION,
+  template: {
+    aliases: {
+      [DEFINITIONS_ALIAS]: {},
+    },
+    mappings: {
```

Comment:
the type_id is used for search is one of the query - should we turn off dynamic mapping and explicitly map the field ? I think we'll have to use dsl instead of esql if we do that otherwise unmapped fields won't show up

Reply:
Fixed in https://github.com/elastic/kibana/pull/201470/commits/e9c2f376472a7d24cd7d4fd27e09fad827eaf90f by:
1. Disabling dynamic mapping
2. Mapping source.type_id
3. Reading all definitions from _source instead of columns (but still via ES|QL)